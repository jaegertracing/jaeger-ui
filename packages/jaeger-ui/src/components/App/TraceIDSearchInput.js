// @flow

// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import { Form, Input } from 'antd';
import { withRouter } from 'react-router-dom';

import type { RouterHistory } from 'react-router-dom';

import { getUrl } from '../../components/TracePage/url';

import './TraceIDSearchInput.css';

type Props = {
  history: RouterHistory,
};

class TraceIDSearchInput extends React.PureComponent<Props> {
  props: Props;

  goToTrace = event => {
    event.preventDefault();
    const value = event.target.elements.idInput.value;
    if (value) {
      this.props.history.push(getUrl(value));
    }
  };

  render() {
    return (
      <Form layout="horizontal" onSubmit={this.goToTrace} className="TraceIDSearchInput--form">
        <Input name="idInput" placeholder="Lookup by Trace ID..." />
      </Form>
    );
  }
}

export default withRouter(TraceIDSearchInput);
