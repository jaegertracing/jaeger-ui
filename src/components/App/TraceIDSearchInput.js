// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';

class TraceIDSearchInput extends Component {
  goToTrace(e) {
    this.props.history.push(`/trace/${this.traceIDInput.value}`);
    e.preventDefault();
    return false;
  }
  render() {
    return (
      <form onSubmit={e => this.goToTrace(e)}>
        <input
          type="text"
          placeholder="Lookup by Trace ID..."
          ref={input => {
            this.traceIDInput = input;
          }}
        />
      </form>
    );
  }
}

TraceIDSearchInput.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
};

export default withRouter(TraceIDSearchInput);
