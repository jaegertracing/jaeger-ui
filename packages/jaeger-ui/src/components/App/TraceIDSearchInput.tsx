// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import { Form, Input, Icon } from 'antd';
import { RouteComponentProps, Router as RouterHistory, withRouter } from 'react-router-dom';

import { getUrl } from '../TracePage/url';

import './TraceIDSearchInput.css';

type Props = RouteComponentProps<any> & {
  history: RouterHistory;
};

class TraceIDSearchInput extends React.PureComponent<Props> {
  goToTrace = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const target = event.target as any;
    const value = target.elements.idInput.value;
    if (value) {
      this.props.history.push(getUrl(value));
    }
  };

  render() {
    return (
      <Form layout="horizontal" onSubmit={this.goToTrace} className="TraceIDSearchInput--form">
        <Input
          autosize={null}
          name="idInput"
          placeholder="Lookup by Trace ID..."
          prefix={<Icon type="search" />}
        />
      </Form>
    );
  }
}

export default withRouter(TraceIDSearchInput);
