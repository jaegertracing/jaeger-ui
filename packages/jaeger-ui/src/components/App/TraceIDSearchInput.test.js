/**
 * Copyright (c) 2020 The Jaeger Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { mount } from 'enzyme';

import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import TraceIDSearchInput from './TraceIDSearchInput';

describe('<TraceIDSearchInput />', () => {
  let wrapper;
  let history;

  beforeEach(() => {
    history = createMemoryHistory();
    wrapper = mount(
      <Router history={history}>
        <TraceIDSearchInput />
      </Router>
    ).find('form');
  });

  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('pushes input id to history', () => {
    const idInput = wrapper.find('input');
    const traceId = 'MOCK-TRACE-ID';
    idInput.instance().value = traceId;

    wrapper.simulate('submit');

    expect(history.length).toEqual(2);
    expect(history.location.pathname).toEqual(`/trace/${traceId}`);
  });

  it('does not push to history on falsy input value', () => {
    wrapper.simulate('submit');

    expect(history.length).toEqual(1);
  });
});
