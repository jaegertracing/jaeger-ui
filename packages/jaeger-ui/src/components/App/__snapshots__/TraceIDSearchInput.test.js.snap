// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TraceIDSearchInput /> renders as expected 1`] = `
<form
  className="ant-form ant-form-horizontal TraceIDSearchInput--form"
  onSubmit={[Function]}
>
  <input
    autosize={null}
    className="ant-input"
    disabled={false}
    name="idInput"
    onKeyDown={[Function]}
    placeholder="Lookup by Trace ID..."
    type="text"
  />
</form>
`;
